datadog_lambda/cold_start.py

```diff
@@ -70,10 +76,8 @@ def __init__(self, module_name, full_file_path, start_time_ns, end_time_ns=None)
 
 
 def reset_node_stacks():
-    global root_nodes
-    root_nodes = []
-    global import_stack
-    import_stack = []
+    root_nodes.clear()
```

Comment:
why are we removing the `globals`?

Reply:
Good question. Replacing the lists with new ones was causing my test to fail because somewhere we're holding onto an old reference to these objects. The correct solution is to just remove all the stuff from inside the list.

Reply:
Yet, this would create an O(n) complexity instead of O(1) of reassigning right? Although I guess we'd have the remaining O(n) space in the wild while reassigning. I tried researching which was the best solution for this, but theres like 5 ways of doing it. So I'll take this one.

Reply:
This method does not do an allocation where as the previous did.